lib/aws-xray-sdk/plugins/ecs.rb

```diff
@@ -10,12 +10,77 @@ module ECS
 
       ORIGIN = 'AWS::ECS::Container'.freeze
 
+      # Only compatible with v4!
+      # The v3 metadata url does not contain cloudwatch informations
+      METADATA_ENV_KEY = 'ECS_CONTAINER_METADATA_URI_V4'
+
       def self.aws
-        @@aws ||= begin
-          { ecs: { container: Socket.gethostname } }
+        metadata = get_metadata()
+
+        begin
+          metadata[:ecs][:container] = Socket.gethostname
         rescue StandardError => e
-          @@aws = {}
           Logging.logger.warn %(cannot get the ecs container hostname due to: #{e.message}.)
+          metadata[:ecs][:container] = nil
+        end
+
+        @@aws = {
+          ecs: metadata[:ecs],
+          cloudwatch_logs: metadata[:cloudwatch_logs]
+        }
+      end
+
+      private
+
+      def self.get_metadata()
+        begin
+          metadata_uri = URI(ENV[METADATA_ENV_KEY])
+          req = Net::HTTP::Get.new(metadata_uri)
+          metadata_json = do_request(req)
+          return parse_metadata(metadata_json)
+        rescue StandardError => e
+          Logging.logger.warn %(cannot get the ecs instance metadata due to: #{e.message}.)
+          { ecs: {}, cloudwatch_logs: {} }
+        end
+      end
+
+      def self.parse_metadata(json_str)
+        data = JSON(json_str)
+
+        metadata = {
+          ecs: {
+            container_arn: data['ContainerARN'],
+          },
+          cloudwatch_logs: {
+            log_group: data["LogOptions"]['awslogs-group'],
+            log_region: data["LogOptions"]['awslogs-region'],
+            arn: data['ContainerARN']
+          }
+        }
```

Comment:
Potentially, much more could be added to the x-ray segment.

This particular information for example would be valuable to hunt regressions:
```
com.amazonaws.ecs.task-definition-version
```

<img width="1306" alt="Screenshot 2023-09-11 at 12 52 41" src="https://github.com/aws/aws-xray-sdk-ruby/assets/1692632/2ccdd31b-a5b2-4573-922d-562dbe78079e">